packages/browser/src/extensions/conversations/external/index.tsx

```diff
@@ -0,0 +1,616 @@
+
+        // Render the widget
+        this._renderWidget(initialState, initialUserTraits)
+
+        // Track widget initialization
+        this._posthog.capture('$conversations_widget_loaded', {
+            hasExistingTicket: !!this._currentTicketId,
+            initialState: initialState,
+            hasUserTraits: !!initialUserTraits,
+        })
+
+        // If we have a ticket, load its messages
+        if (this._currentTicketId) {
+            this._loadMessages()
+        }
+
+        // Start polling for messages (always, to show unread badge)
+        this._startPolling()
+
+        // Listen for identify events to handle distinct_id changes
+        this._setupIdentifyListener()
+    }
+
+    /**
+     * Get initial user traits from PostHog or localStorage
+     */
+    private _getInitialUserTraits(): UserProvidedTraits | null {
+        // First, check PostHog's person properties
+        const personProperties = this._posthog.persistence?.props || {}
+        const posthogName = personProperties.$name || personProperties.name
+        const posthogEmail = personProperties.$email || personProperties.email
+
+        // If we have traits from PostHog, use those
+        if (posthogName || posthogEmail) {
+            return {
+                name: posthogName || undefined,
+                email: posthogEmail || undefined,
+            }
+        }
+
+        // Otherwise, check localStorage for previously saved traits
+        const savedTraits = this._persistence.loadUserTraits()
+        if (savedTraits && (savedTraits.name || savedTraits.email)) {
+            return savedTraits
+        }
+
+        return null
+    }
+
+    /**
+     * Handle user identification from the widget form
+     */
+    private _handleIdentify = (traits: UserProvidedTraits): void => {
+        // Save traits to localStorage
+        this._persistence.saveUserTraits(traits)
+
+        // Track identification
+        this._posthog.capture('$conversations_user_identified', {
+            hasName: !!traits.name,
+            hasEmail: !!traits.email,
+        })
+    }
+
+    /**
+     * Handle sending a message
+     */
+    private _handleSendMessage = async (message: string): Promise<void> => {
+        // Get user traits from the widget
+        const userTraits = this._widgetRef?.getUserTraits() || undefined
+
+        const isNewTicket = !this._currentTicketId
+
+        try {
+            // Call API directly
+            const response = await this._apiSendMessage(message, this._currentTicketId || undefined, userTraits)
+
+            // Update current ticket ID
+            if (!this._currentTicketId) {
+                this._currentTicketId = response.ticket_id
+                this._persistence.saveTicketId(response.ticket_id)
+                logger.info('New ticket created', { ticketId: response.ticket_id })
+            }
+
+            // Track message sent
+            this._posthog.capture('$conversations_message_sent', {
```

Comment:
we should _maybe_ exclude these `$conversations_etc` events from billing  
and then set up tracking in the billing report  
so that this is free while in beta  
  
but it's not going to be _sooo_ many events so maybe OK to have it be the event cost for now